sdk/core/Azure.Core/tests/RecordSessionTests.cs

```diff
@@ -409,6 +409,39 @@ public void RecordSessionLookupSkipsRequestBodyWhenFilterIsOn()
             Assert.Throws<TestRecordingMismatchException>(() => playbackTransport.Process(message));
         }
 
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public void ContentLengthUpdatedCorrectlyOnEmptyBody(bool isHeadRequest)
```

Comment:
extra super nit: I would not make this test parameterized and instead had two separate ones.

Reply:
I can make two separate test wrappers and leave this as an internal helper.